src/components/ContentNode/Grid.vue

```diff
@@ -0,0 +1,55 @@
+<!--
+  This source file is part of the Swift.org open source project
+
+  Copyright (c) 2022 Apple Inc. and the Swift project authors
+  Licensed under Apache License v2.0 with Runtime Library Exception
+
+  See https://swift.org/LICENSE.txt for license information
+  See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+<template>
+  <div class="grid" :style="style" :class="{ 'with-columns': columns }">
+    <slot />
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'grid',
+  props: {
+    columns: {
+      type: Number,
+      default: null,
+      required: false,
+      validator: v => v > 0,
+    },
+  },
+  computed: {
+    style: ({ columns }) => ({ '--col-count': columns }),
+  },
+};
+</script>
+
+<style scoped lang='scss'>
+@import 'docc-render/styles/_core.scss';
+
+.grid {
```

Comment:
I think you'll need to add some basic spacing rules for this new block element to more closely match the spacing that we have for paragraphs and other things.

Right now, this is what it looks like when I splice your example JSON into an existing doc—notice how the paragraph is squeezed against the grid block.

<img width="674" alt="Screenshot 2022-08-25 at 3 40 41 PM" src="https://user-images.githubusercontent.com/212918/186781619-7cfde647-55a2-4f98-b1f2-e0cd1abe12f3.png">

I think you can probably just match what we're doing for a similar thing like code listings—I imagine it uses the `$stacked-margin-large` variable or something similar using a generic sibling selector.

Reply:
Like this I think: https://github.com/apple/swift-docc-render/blob/9a608c3d457cb6c7880773941ffac2a8c792b97f/src/components/DocumentationTopic/ContentNode.vue#L43